File: src/main/java/no/brreg/organizationcatalogue/model/OrganizationDB.java
package no.brreg.organizationcatalogue.model;

import javax.validation.constraints.NotBlank;

import no.brreg.organizationcatalogue.generated.model.PrefLabel;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDate;

@Document(collection="organizations")
public class OrganizationDB {
    @Id private String organizationId;
    @NotBlank private String name;
    private String internationalRegistry;
    private String orgType;
    private String orgPath;
    private String subOrganizationOf;
    private LocalDate issued;
    private String municipalityNumber;
    private String industryCode;
    private String sectorCode;
    private PrefLabel prefLabel;

    public String getOrganizationId() {
        return organizationId;
    }

    public void setOrganizationId(String organizationId) {
        this.organizationId = organizationId;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getInternationalRegistry() {
        return internationalRegistry;
    }

    public void setInternationalRegistry(String internationalRegistry) {
        this.internationalRegistry = internationalRegistry;
    }

    public String getOrgType() {
        return orgType;
    }

    public void setOrgType(String orgType) {
        this.orgType = orgType;
    }

    public String getOrgPath() {
        return orgPath;
    }

    public void setOrgPath(String orgPath) {
        this.orgPath = orgPath;
    }

    public String getSubOrganizationOf() {
        return subOrganizationOf;
    }

    public void setSubOrganizationOf(String subOrganizationOf) {
        this.subOrganizationOf = subOrganizationOf;
    }

    public LocalDate getIssued() {
        return issued;
    }

    public void setIssued(LocalDate issued) {
        this.issued = issued;
    }

    public String getMunicipalityNumber() {
        return municipalityNumber;
    }

    public void setMunicipalityNumber(String municipalityNumber) {
        this.municipalityNumber = municipalityNumber;
    }

    public String getIndustryCode() {
        return industryCode;
    }

    public void setIndustryCode(String industryCode) {
        this.industryCode = industryCode;
    }

    public String getSectorCode() {
        return sectorCode;
    }

    public void setSectorCode(String sectorCode) {
        this.sectorCode = sectorCode;
    }

    public PrefLabel getPrefLabel() {
        return prefLabel;
    }

    public void setPrefLabel(PrefLabel prefLabel) {
        this.prefLabel = prefLabel;
    }
}

File: src/main/kotlin/no/digdir/organizationcatalog/model/OrganizationDB.kt
package no.digdir.organizationcatalog.model

import jakarta.persistence.Column
import jakarta.persistence.Embeddable
import jakarta.persistence.Embedded
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.Id
import jakarta.persistence.Table
import jakarta.validation.constraints.NotBlank
import java.time.LocalDate

@Embeddable
data class EmbeddedPrefLabel(
    @Column(name = "pref_label_nb")
    val nb: String? = null,
    @Column(name = "pref_label_nn")
    val nn: String? = null,
    @Column(name = "pref_label_en")
    val en: String? = null,
)

@Entity
@Table(name = "organizations")
data class OrganizationDB(
    @Id
    @Column(name = "organization_id")
    val organizationId: String = "",
    @field:NotBlank
    @Column(name = "name", nullable = false)
    val name: String = "",
    @Column(name = "international_registry")
    val internationalRegistry: String? = null,
    @Column(name = "org_type")
    val orgType: String? = null,
    @Column(name = "org_path")
    val orgPath: String? = null,
    @Column(name = "sub_organization_of")
    val subOrganizationOf: String? = null,
    @Column(name = "issued")
    val issued: LocalDate? = null,
    @Column(name = "municipality_number")
    val municipalityNumber: String? = null,
    @Column(name = "industry_code")
    val industryCode: String? = null,
    @Column(name = "sector_code")
    val sectorCode: String? = null,
    @Embedded
    val prefLabel: EmbeddedPrefLabel? = null,
    @Enumerated(EnumType.STRING)
    @Column(name = "org_status")
    val orgStatus: OrgStatus? = null,
    @Column(name = "homepage")
    val homepage: String? = null,
    @Column(name = "subordinate", nullable = false)
    val subordinate: Boolean = false,
)
